src/components/Sections/hooks/useKonnectorsByCat.ts

```diff
@@ -1,75 +1,44 @@
 import { sortBy } from 'lodash'
 import { useSelector } from 'react-redux'
-import { useState, useEffect, useMemo } from 'react'
-
-import type { IOCozyApp, IOCozyKonnector } from 'cozy-client/types/types'
+import { useState, useEffect, useMemo, useCallback } from 'react'
+
+import type {
+  IOCozyAccount,
+  IOCozyApp,
+  IOCozyKonnector,
+  IOCozyTrigger
+} from 'cozy-client/types/types'
 import { useClient, useQuery, useAppsInMaintenance } from 'cozy-client'
+import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
 
 import { Section } from 'components/Sections/SectionsTypes'
 import { fetchAllKonnectors } from 'components/Sections/queries/konnectors'
 import { getInstalledKonnectors } from 'selectors/konnectors'
 import { suggestedKonnectorsConn } from 'queries'
+import { formatServicesSections } from 'components/Sections/services/formatServicesSections'
 
-const transformAndSortData = (
-  data: { [key: string]: IOCozyKonnector[] },
-  installedKonnectors: IOCozyKonnector[],
-  suggestedKonnectors: IOCozyKonnector[],
-  appsAndKonnectorsInMaintenance: IOCozyKonnector[]
-): Section[] => {
-  const installedKonnectorNames = new Set(installedKonnectors.map(k => k.name))
-  const maintenanceSlugs = new Set(
-    appsAndKonnectorsInMaintenance.map(k => k.slug)
-  )
-
-  const sections = Object.keys(data).map(key => {
-    const allItems = data[key] || []
-    const availableItems = allItems.filter(
-      item => !maintenanceSlugs.has(item.slug)
-    )
-    const installedItems = availableItems.filter(item =>
-      installedKonnectorNames.has(item.name)
-    )
-    const suggestedItems = availableItems.filter(item =>
-      suggestedKonnectors.some(k => k.slug === item.slug)
-    )
-    const items =
-      installedItems.length > 0
-        ? [...installedItems, ...suggestedItems]
-        : availableItems
-
-    return {
-      name: key,
-      items,
-      id: key,
-      type: 'category',
-      layout: {
-        originalName: key,
-        createdByApp: '',
-        mobile: { detailedLines: false, grouped: true },
-        desktop: { detailedLines: false, grouped: true },
-        order: 0
-      },
-      pristine: installedItems.length === 0
-    }
-  })
+export const useKonnectorsByCat = (): Section[] => {
+  const client = useClient()
+  const { t } = useI18n()
 
-  sections.sort((a, b) => {
-    if (!a.pristine && b.pristine) {
-      return -1
-    }
-    if (a.pristine && !b.pristine) {
-      return 1
-    }
-    return a.name.localeCompare(b.name)
-  })
+  // Unsure how to handle the deps issues, useSelector should already be memoized
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const allTriggers =
+    useSelector(
```

Comment:
Why do you use a useSelector rather than a useQuery with a query that retrieves everything?

Reply:
updated